android/src/main/java/io/sentry/RNSentryModule.java

```diff
@@ -1,34 +1,265 @@
+                            null,
+                            event.getMap("user").getString("email")
+                    )
+            );
+
+            helper.helpBuildingEvent(eventBuilder);
+
+            if (this.extra != null) {
+                for (Map.Entry<String, Object> entry : recursivelyDeconstructReadableMap(this.extra).entrySet()) {
+                    eventBuilder.withExtra(entry.getKey(), entry.getValue());
+                }
+            }
+            for (Map.Entry<String, Object> entry : recursivelyDeconstructReadableMap(event.getMap("extra")).entrySet()) {
+                eventBuilder.withExtra(entry.getKey(), entry.getValue());
+            }
+            
+            if (this.tags != null) {
+                for (Map.Entry<String, Object> entry : recursivelyDeconstructReadableMap(this.tags).entrySet()) {
+                    eventBuilder.withExtra(entry.getKey(), entry.getValue());
+                }
+            }
+            for (Map.Entry<String, Object> entry : recursivelyDeconstructReadableMap(event.getMap("tags")).entrySet()) {
+                eventBuilder.withTag(entry.getKey(), entry.getValue().toString());
+            }
+            
+            Sentry.capture(eventBuilder.build());
+        } else {
+            logger.info("Event has no key message which means it is a js error");
+        }
+    }
+
+    @ReactMethod
+    public void clearContext() {
+        Sentry.clearContext();
+    }
+
+    @ReactMethod
+    public void activateStacktraceMerging(Promise promise) {
+        logger.info("TODO: implement activateStacktraceMerging");
+        promise.resolve(true);
+    }
+
+    private Breadcrumb.Level breadcrumbLevel(String level) {
+        switch (level) {
+            case "critical":
+                return Breadcrumb.Level.CRITICAL;
+            case "warning":
+                return Breadcrumb.Level.WARNING;
+            case "info":
+                return Breadcrumb.Level.INFO;
+            case "debug":
+                return Breadcrumb.Level.DEBUG;
+            default:
+                return Breadcrumb.Level.ERROR;
+        }
+    }
+
+    private Event.Level eventLevel(String level) {
+        switch (level) {
+            case "fatal":
+                return Event.Level.FATAL;
+            case "warning":
+                return Event.Level.WARNING;
+            case "info":
+                return Event.Level.INFO;
+            case "debug":
+                return Event.Level.DEBUG;
+            default:
+                return Event.Level.ERROR;
+        }
+    }
+
+    private Level logLevel(int level) {
+        switch (level) {
+            case 1:
+                return Level.SEVERE;
+            case 2:
+                return Level.INFO;
+            case 3:
+                return Level.ALL;
+            default:
+                return Level.OFF;
+        }
+    }
+
+    private Map<String, Object> recursivelyDeconstructReadableMap(ReadableMap readableMap) {
```

Comment:
If this can be typed as `ReadableNativeMap`, then you can use [toHashMap](https://github.com/facebook/react-native/blob/976abf87f213eb3a05277a9956b5e582ba951a2c/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableNativeMap.java#L60)